core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go

```diff
@@ -774,9 +800,526 @@ func setupSimulatedBackendAndAuth(t *testing.T) (*simulated.Backend, *bind.Trans
+func Benchmark_CCIPReader_CommitReportsGTETimestamp(b *testing.B) {
+	tests := []struct {
+		logsInsertedFirst        int
+		logsInsertedToMatchQuery int
+	}{
+		{0, 0},
+		{1, 10},
+		{10, 100},
+		{100, 10_000},
+	}
+
+	for _, tt := range tests {
+		b.Run(fmt.Sprintf("FirstLogs_%d_MatchLogs_%d", tt.logsInsertedFirst, tt.logsInsertedToMatchQuery), func(b *testing.B) {
+			benchmarkCommitReports(b, tt.logsInsertedFirst, tt.logsInsertedToMatchQuery)
+		})
+	}
+}
+
+func benchmarkCommitReports(b *testing.B, logsInsertedFirst int, logsInsertedToMatchQuery int) {
+	ctx := testutils.Context(b)
+	sb, auth := setupSimulatedBackendAndAuth(b)
+
+	// Initialize test setup
+	ts := testSetup(
+		ctx,
+		b,
+		chainS1,
+		chainS1,
+		nil,
+		evmconfig.DestReaderConfig,
+		nil,
+		nil,
+		false,
+		sb,
+		auth,
+		true)
+
+	err := ts.extendedCR.Bind(ctx, []types.BoundContract{
+		{
+			Address: ts.contractAddr.String(),
+			Name:    consts.ContractNameOffRamp,
+		},
+	})
+	require.NoError(b, err)
+
+	// Insert logs in two phases based on parameters
+	if logsInsertedFirst > 0 {
+		populateDatabaseForCommitReportAccepted(b, ts, chainS1, logsInsertedFirst, 0)
+	}
+
+	queryTimestamp := time.Now()
+
+	if logsInsertedToMatchQuery > 0 {
+		populateDatabaseForCommitReportAccepted(b, ts, chainS1, logsInsertedToMatchQuery, logsInsertedFirst)
+	}
+
+	// Reset timer to measure only the query time
+	b.ResetTimer()
+
+	// Execute the query
+	_, err = ts.extendedCR.ExtendedQueryKey(
+		ctx,
+		consts.ContractNameOffRamp,
+		query.KeyFilter{
+			Key: consts.EventNameCommitReportAccepted,
+			Expressions: []query.Expression{
+				// #nosec G115
+				query.Timestamp(uint64(queryTimestamp.Unix()), primitives.Gte), // Safe: offset and i are always non-negative and within range
+				query.Confidence(primitives.Finalized),
+			},
+		},
+		query.LimitAndSort{
+			SortBy: []query.SortBy{query.NewSortByTimestamp(query.Asc)},
+		},
+		&offramp.OffRampCommitReportAccepted{},
+	)
+	require.NoError(b, err)
+}
+
+func populateDatabaseForCommitReportAccepted(
+	b *testing.B,
+	testEnv *testSetupData,
+	sourceChain cciptypes.ChainSelector,
+	numOfReports int,
+	offset int,
+) {
+	ctx := testutils.Context(b)
+
+	var logs []logpoller.Log
+	commitReportEvent, exists := offrampABI.Events[consts.EventNameCommitReportAccepted]
+	require.True(b, exists, "Event CommitReportAccepted not found in ABI")
+
+	commitReportEventSig := commitReportEvent.ID
+	commitReportAddress := testEnv.contractAddr
+
+	for i := 0; i < numOfReports; i++ {
+		// Calculate unique BlockNumber and LogIndex
+		blockNumber := int64(offset + i + 1) // Offset ensures unique block numbers
+		logIndex := int64(offset + i + 1)    // Offset ensures unique log indices
+
+		// Simulate merkleRoots
+		merkleRoots := []offramp.InternalMerkleRoot{
+			{
+				SourceChainSelector: uint64(sourceChain),
+				OnRampAddress:       []byte{0x12, 0x34},
```

Comment:
There should be an util for generating random address, which might be a better approach
```go
onRampAddress := utils.RandomAddress()
OnRampAddress: onrampAddress
```